src/adapter/src/catalog.rs

```diff
@@ -163,7 +163,7 @@ pub const LINKED_CLUSTER_REPLICA_NAME: &str = "linked";
 pub struct Catalog {
     state: CatalogState,
     plans: CatalogPlans,
-    storage: Arc<tokio::sync::Mutex<mz_catalog::Connection>>,
+    storage: Arc<tokio::sync::Mutex<Box<dyn mz_catalog::DurableCatalogState>>>,
```

Comment:
```
pub struct Catalog<S = mz_catalog::stash::Connection> {
    storage: Arc<tokio::sync::Mutex<S>>,
    ..
}
```

Thoughts about making Catalog generic over storage, then doing

```
impl<S: mz_catalog::DurableCatalogState> Catalog<S> {
```

below? I don't have strong opinions either way.

Reply:
I'll play around with this, I don't have a strong opinion but went with the `Box<dyn ...>` approach because I just found it much easier to work with.

Reply:
I think I'm going to keep the `Box<dyn ...>` for now. The generic approach with a trait bound is really intrusive into all the adapter code. I also think (hope) dynamic dispatch isn't going to be the bottleneck on catalog transactions.